tools/walletextension/wallet_extension.go

```diff
@@ -478,3 +478,19 @@ func (w *WalletExtension) Version() string {
 func (w *WalletExtension) GetTenNodeHealthStatus() (bool, error) {
 	return w.tenClient.Health()
 }
+
+func (w *WalletExtension) GetMessage(encryptionToken string, formatsSlice []string) (string, error) {
+	// Check if the formats are valid
+	for _, format := range formatsSlice {
+		if _, exists := common.SignatureTypeMap[format]; !exists {
+			return "", fmt.Errorf("invalid format: %s", format)
+		}
+	}
+
+	messageFormat := common.GetBestFormat(formatsSlice)
+	message, err := viewingkey.GenerateMessage(encryptionToken, int64(w.config.TenChainID), viewingkey.PersonalSignMessageSupportedVersions[0], messageFormat, false)
```

Comment:
I don't understand the logic around `PersonalSignMessageSupportedVersions`

Reply:
It was intended if we would want to support multiple versions in the future, but it is unnecessary complication for now.
Removed and used a constant insted. When/if we add another version for some reason we can do it that way.